Added info on dependency version constraints (#5431)
dart-lang/site-www: src/content/tools/pub/dependencies.md

aren't on the latest stable versions,
use [`dart pub outdated`][].

### Tighten version constraints for dev dependencies

A dev dependency defines a package that you need only when developing.
A finished app won't need these packages.
Examples of these packages include tests or code generation tooling.
Set the version constraints of packages in [`dev_dependencies`][dev-dep]
to have a lower bound of the latest version on which your package depends.

Tightening the version constraints of your dev dependencies might
resemble the following:

```yaml
dev_dependencies:
  build_runner: ^2.4.8
  lints: ^2.1.1
  test: ^1.25.1
```

> This YAML sets the `dependencies` to the major version and the
> `dev_dependencies` to a specific patch version.

Reviewer: Since you dropped the `dependencies` from the example, I guess need to modify this text:

```suggestion
This YAML sets the `dev_dependencies` to the latest patch versions.
```
